planner/funcdep/fd_graph.go

```diff
@@ -879,7 +882,7 @@ func (s FDSet) AllCols() FastIntSet {
 // AddFrom merges two FD sets by adding each FD from the given set to this set.
 // Since two different tables may have some column ID overlap, we better use
 // column unique ID to build the FDSet instead.
-func (s *FDSet) AddFrom(fds *FDSet) {
+func (s *FDSet) AddFrom(fds *FDSet, acrossBlock bool) {
```

Comment:
Add some comments about the new arg `acrossBlock`

Reply:
// AddFrom is commonly used to pull underlining FD up to current operator. Since
// some ancillary factors for FD like hasAggBuild and GroupCols is strictly limited
// into an exact scope --- that logical query block, but FD doesn't. So we should
// clean that stuff when pulling FD across logical query blocks.